src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/SimdVectorExtensions.cs

```diff
@@ -0,0 +1,171 @@
+        public static void CopyTo<TVector, T>(this TVector vector, T[] destination, int startIndex)
+            where TVector : ISimdVector<TVector, T>
+        {
+            TVector.CopyTo(vector, destination, startIndex);
+        }
+
+        /// <summary>Copies a vector to a given span.</summary>
+        /// <typeparam name="TVector">The type of the vector.</typeparam>
+        /// <typeparam name="T">The type of the elements in the vector.</typeparam>
+        /// <param name="vector">The vector to be copied.</param>
+        /// <param name="destination">The span to which the <paramref name="vector" /> is copied.</param>
+        /// <exception cref="ArgumentException">The length of <paramref name="destination" /> is less than <see cref="ISimdVector{TVector, T}.Count" />.</exception>
+        /// <exception cref="NotSupportedException">The type of the elements in the vector (<typeparamref name="T" />) is not supported.</exception>
+        public static void CopyTo<TVector, T>(this TVector vector, Span<T> destination)
+            where TVector : ISimdVector<TVector, T>
+        {
+            TVector.CopyTo(vector, destination);
+        }
+
+        /// <summary>Gets the element at the specified index.</summary>
+        /// <typeparam name="TVector">The type of the vector.</typeparam>
+        /// <typeparam name="T">The type of the elements in the vector.</typeparam>
+        /// <param name="vector">The vector to get the element from.</param>
+        /// <param name="index">The index of the element to get.</param>
+        /// <returns>The value of the element at <paramref name="index" />.</returns>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="index" /> was less than zero or greater than the number of elements.</exception>
+        /// <exception cref="NotSupportedException">The type of <paramref name="vector" /> (<typeparamref name="T" />) is not supported.</exception>
+        public static T GetElement<TVector, T>(this TVector vector, int index)
+            where TVector : ISimdVector<TVector, T>
+        {
+            return TVector.GetElement(vector, index);
+        }
+
+#pragma warning disable CS8500 // This takes the address of, gets the size of, or declares a pointer to a managed type ('T')
+        /// <summary>Stores a vector at the given destination.</summary>
+        /// <typeparam name="TVector">The type of the vector.</typeparam>
+        /// <typeparam name="T">The type of the elements in the vector.</typeparam>
+        /// <param name="source">The vector that will be stored.</param>
+        /// <param name="destination">The destination at which <paramref name="source" /> will be stored.</param>
+        /// <exception cref="NotSupportedException">The type of <paramref name="source" /> (<typeparamref name="T" />) is not supported.</exception>
+        public static void Store<TVector, T>(this TVector source, T* destination)
```

Comment:
Are these extensions here because the corresponding methods are statics defined not on the generic `VectorXx<T>` but instead on the non-generic `VectorXx` types?

Reply:
Same as above on `GetElement`. It follows our existing pattern so that core APIs can appear like instance methods but avoid requiring additional locals and passing the `this` parameter `byref`.